hazelcast/src/main/java/com/hazelcast/jet/impl/operation/AbstractJobOperation.java

```diff
@@ -58,4 +59,12 @@ protected void readInternal(ObjectDataInput in) throws IOException {
         super.readInternal(in);
         jobId = in.readLong();
     }
+
+    protected JetServiceBackend getJetServiceBackend() {
+        if (!getNodeEngine().getConfig().getJetConfig().isEnabled()) {
+            throw new IllegalArgumentException("Jet is disabled, see JetConfig#setEnabled.");
```

Comment:
One minor suggestion, since I saw elsewhere that Can said:
> Can Gencer
> it would be good to refer to non-programmatic config as well in this kind of errors
> also it's not clear that it's referring to a server-side config

Can we add something like "The jet service (engine) is disabled, please enable it in your member configuration"? We can refer to the documentation if we want: "Please refer to the Hazelcast documentation for more information.".

Albeit, this is already much better than what we had before, so maybe it's good enough for BETA :) And if you push a commit, it will restart the PR builder, and we don't want that :)

Reply:
Similarly to what I suggested for the log message, we should be more explicit in what to do to fix it - see https://github.com/hazelcast/hazelcast/pull/19050/files

Reply:
Just leave it as-is so the PR builder finishes and @gurbuzali if you have a better message, send it to me and I'll push directly to master :)